Dockerfile

```diff
@@ -1,41 +1,45 @@
+ARG HOMEDIR=/opt/gatekeeper
+
 #
-# Builder image
+# Builder
 #
 
 FROM golang:1.14.4 AS build-env
 ARG SOURCE=*
+ARG HOMEDIR
 
 ADD $SOURCE /src/
 WORKDIR /src/
 
-# Unpack any tars, then try to execute a Makefile, but if the SOURCE url is
-# just a tar of binaries, then there probably won't be one. Using multiple RUN
-# commands to ensure any errors are caught.
-RUN find . -name '*.tar.gz' -type f | xargs -rn1 tar -xzf
-RUN if [ -f Makefile ]; then make; fi
-RUN cp "$(find . -name 'gatekeeper' -type f -print -quit)" /gatekeeper
+RUN make static
+
+WORKDIR ${HOMEDIR}
+
+RUN cp /src/bin/gatekeeper .
+COPY templates ./templates
+
+RUN echo "gatekeeper:x:1000:gatekeeper" >> /etc/group && \
+    echo "gatekeeper:x:1000:1000:gatekeeper user:${HOMEDIR}:/sbin/nologin" >> /etc/passwd && \
+    chown -R gatekeeper:gatekeeper ${HOMEDIR} && \
+    chmod -R g+rw ${HOMEDIR} && \
+    chmod +x gatekeeper
 
 #
 # Actual image
 #
 
-FROM registry.access.redhat.com/ubi8/ubi-minimal:8.3
+FROM scratch
+ARG HOMEDIR
 
 LABEL Name=gatekeeper \
       Release=https://github.com/gogatekeeper/gatekeeper \
       Url=https://github.com/gogatekeeper/gatekeeper \
       Help=https://github.com/gogatekeeper/gatekeeper/issues
 
-WORKDIR "/opt/gatekeeper"
-
-RUN echo "gatekeeper:x:1000:gatekeeper" >> /etc/group && \
-    echo "gatekeeper:x:1000:1000:gatekeeper user:/opt/gatekeeper:/sbin/nologin" >> /etc/passwd && \
-    chown -R gatekeeper:gatekeeper /opt/gatekeeper && \
-    chmod -R g+rw /opt/gatekeeper
-
-COPY templates ./templates
-COPY --from=build-env /gatekeeper ./
-RUN chmod +x gatekeeper
+COPY --from=build-env ${HOMEDIR} ${HOMEDIR}
+COPY --from=build-env /etc/passwd /etc/passwd
+COPY --from=build-env /etc/group /etc/group
```

Comment:
I would copy also CA certificates

Reply:
yup i can copy it there as default, although in serious deployment you would probably need to have mounted some fresh CA's